message TestProto {
  optional string test_string = 1;
  optional uint64 test_uint = 2;
  message SubProto {
    optional uint64 id = 1;
    optional double num = 2;
  }
  optional SubProto sub_message = 3;
  optional bool is_sane = 4;
}
